galite-core/src/main/kotlin/org/kopi/galite/domain/Domain.kt

```diff
@@ -25,6 +25,12 @@ package org.kopi.galite.domain
  */
 open class Domain<T: Comparable<T>>(val length: Int? = null) {
 
-    /** Override it if you want to check values when they are passed. */
+    /** Override it if you want to define a constraint that the domain values ​​must meet. */
     open val check: ((value: T) -> Boolean)? = null
+
+    /**
+     * Override it if you want to apply transformation on values.
```

Comment:
It is mentioned that with the assignment of a transformation like Transformation.convertUpper to the transformation variable allows it to be applied, while you have to call the convertUpper method which allows the conversion in init {} of the field class for example.

Reply:
I think we should avoid using init{} in domains

Reply:
The convertUpper should be initialized in Field.kt not in domain. In the other case, the comment must be changed because the affectation of the transformation don't mean apply it.